mysql-test/suite/tokudb.rpl/t/rpl_tokudb_rfr_partition_table.test

```diff
@@ -0,0 +1,75 @@
+# test tokudb read free replication feature with partition table
+
+--source include/have_tokudb.inc
+--source include/have_binlog_format_row.inc
+--source include/master-slave.inc
+
+call mtr.add_suppression(".*read free replication is disabled for TokuDB table.*continue with rows lookup");
+
+connection master;
+
+# partition table with explicit PK
+CREATE TABLE t1 (id int(11) NOT NULL, pid int(11), PRIMARY KEY (id)) ENGINE=TokuDB
+PARTITION BY RANGE (id)
+(PARTITION p_1 VALUES LESS THAN (10) ENGINE = TokuDB,
+ PARTITION p_2 VALUES LESS THAN (20) ENGINE = TokuDB,
+ PARTITION p_all VALUES LESS THAN MAXVALUE ENGINE = TokuDB);
+
+insert into t1 values (1, 1), (2, 2), (3, 3), (11, 11), (12, 12), (13, 13);
+
+# partition table without explicit PK
+CREATE TABLE t2 (id int(11) NOT NULL, pid int(11), key idx_1(id)) ENGINE=TokuDB
+PARTITION BY RANGE (id)
+(PARTITION p_1 VALUES LESS THAN (10) ENGINE = TokuDB,
+ PARTITION p_2 VALUES LESS THAN (20) ENGINE = TokuDB,
+ PARTITION p_all VALUES LESS THAN MAXVALUE ENGINE = TokuDB);
+
+insert into t2 values (1, 1), (2, 2), (3, 3), (11, 11), (12, 12), (13, 13);
+
+--sync_slave_with_master
+
+# set tokudb rfr delay
+connection slave;
+set global tokudb_rpl_unique_checks_delay=10000;
+set global tokudb_rpl_lookup_rows_delay=10000;
+stop slave sql_thread;
+start slave sql_thread;
+
+connection master;
+select unix_timestamp() into @tstart;
+insert into t1 values(21, 21);
+delete from t1 where id = 11;
+update t1 set pid = 2 where id = 1;
+
+sync_slave_with_master;
+
+# make sure no unique checks or row lookups is invoked
+connection master;
+select unix_timestamp() into @tend;
+select @tend-@tstart < 10;
```

Comment:
This is a negative test, the delays should only get hit if RFR is not functioning. Without this, there is no way to know that the actual read free operation took place. DEBUG_SYNC would be ideal here but since we need the sync point activated on the slave SQL thread, there is no way to make use of it.